### PR TITLE
[Distributed] Fix _executeDistributedTarget ABI

### PR DESCRIPTION
**Pull request by ktoso**
We should not change the signature of `_executeDistributedTarget` and we actually do not need to -- we do not use the witness table of actor type at all.

This is a follow up to https://github.com/apple/swift/pull/70842 which too aggressively added that generic parameter to runtime entry point

### CODE REVIEW

File: stdlib/public/Distributed/DistributedActorSystem.swift [hunk 1]
```diff
@@ -652,8 +652,8 @@ public struct RemoteCallTarget: CustomStringConvertible, Hashable {
 
 @available(SwiftStdlib 5.7, *)
 @_silgen_name("swift_distributed_execute_target")
-func _executeDistributedTarget<D: DistributedTargetInvocationDecoder, DA: DistributedActor>(
-  on actor: DA, // DistributedActor
+func _executeDistributedTarget<D: DistributedTargetInvocationDecoder>(
```
Comment (ktoso) [inline stdlib/public/Distributed/DistributedActorSystem.swift:655]: ^ we needed to undo this introduction of the DA parameter here

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Comment by ktoso** [comment]
@swift-ci please test

**Review by xedin — COMMENTED** [review]
Thank you!

**Pull request merged**